src/main/java/com/elytradev/concrete/network/Message.java

```diff
@@ -82,15 +83,25 @@ public Message(NetworkContext ctx) {
 		
 		ClassInfo ci = classInfo.get(getClass());
 		if (ci == null) {
-			ReceivedOn ro = getClass().getDeclaredAnnotation(ReceivedOn.class);
+			ReceivedOn ro = null;
```

Comment:
Should probably make a getDeclaredAnnotation utility within Common and use that instead, to keep the code clean.